### PR TITLE
Update E2E config to support screenshots on test failure

### PR DESCRIPTION
**Pull request by louwie17**
Make use of the `useE2EJestConfig` function so it correctly uses the `e2e-environment` libraries `setupFilesAfterEnv` files.
This also uploads the screenshots if the exist (only on test failure) as artifacts in Github actions.

### Detailed test instructions:

- Make sure that e2e github action passes correctly, here is a sample of a [past run that failed](https://github.com/woocommerce/woocommerce-admin/actions/runs/778144422) with a generated artifact.
- Modify one of the e2e tests to make sure it fails (locally) 
- Run the tests using `npx wc-e2e docker:up && WC_E2E_SCREENSHOTS=1 npx wc-e2e test:e2e`
- Check the `tests/e2e/screenshots` folder, it should contain a screenshot for the failed test.

### CODE REVIEW

**Comment by louwie17** [comment]
@adrianduffell @samueljseay I added a screenshot for when it fails, attaching it to an artifact, and it appears the WP installation didn't finish when it goes to the login page:
login-page-failed 2

**Comment by adrianduffell** [comment]
@louwie17 Nice find! Thanks for investigating this.

It looks like it is being redirected to [`wp-admin/install.php`](https://github.com/WordPress/WordPress/blob/master/wp-admin/install.php).

I think we could solve this in core by adding to the [wait-for-build.sh](https://github.com/woocommerce/woocommerce/blob/0df1d3ef64660be4534572390c5fce5471e39e41/tests/e2e/env/bin/wait-for-build.sh#L14) logic. There could be another conditional check that `wp-login.php` returns a `200` status code.

**Comment by louwie17** [comment]
> It looks like it is being redirected to [`wp-admin/install.php`](https://github.com/WordPress/WordPress/blob/master/wp-admin/install.php).
> 
> I think we could solve this in core by adding to the [wait-for-build.sh](https://github.com/woocommerce/woocommerce/blob/0df1d3ef64660be4534572390c5fce5471e39e41/tests/e2e/env/bin/wait-for-build.sh#L14) logic. There could be another conditional check that `wp-login.php` returns a `200` status code.

@adrianduffell I did reproduce this screen locally, and the `wp-login.php` will actually return a `302`, which the wait-for-build script does catch. I even updated the docker-compose script to delay the install on purpose and see what happens. Locally the `wait-for-build` waited until the installation finished.

I am not sure why this screen might show up then, so leaving the `sleep 10` in. I did however get the `WC_E2E_SCREENSHOTS=1` to work, which should help further debug if problems continue to happen. Each E2E test has a screen shot artifact if they were generated (only on failure will it create a screenshot).

File: tests/e2e/config/jest.config.js [hunk 1]
```diff
@@ -1,8 +1,7 @@
 const path = require( 'path' );
-const { jestConfig: baseE2Econfig } = require( '@woocommerce/e2e-environment' );
+const { useE2EJestConfig } = require( '@woocommerce/e2e-environment' );
```
Comment (louwie17) [inline tests/e2e/config/jest.config.js:2]: The jestConfig that was imported here was really the automattic/puppeteer-utils version, and not the one generated by `e2e-environment` [here](https://github.com/woocommerce/woocommerce/blob/trunk/tests/e2e/env/config/jest.config.js#L59). Using the `useE2EJestConfig` fixes this, and enables the screenshot functionality.

**Review by jeffstieler — APPROVED** [review]
This looks good and tests well. I assume the removal of `wait-for-build` is because it's redundant with `@woocommerce/e2e`? If so, 🚢  it.

**Comment by louwie17** [comment]
> I assume the removal of wait-for-build is because it's redundant with @woocommerce/e2e? 

@jeffstieler correct, it was left over from the last refactor that moved to `npx wc-e2e test:e2e` to run e2e tests.

**Pull request merged**